### ISSUE TITLE
Get properties for a subgroup of atoms

### ISSUE THREAD

**Issue by Luthaf** [Luthaf/cymbalum#20]
Users should be able to get some property for a subgroup of particles: a molecule, all waters, all oxygens, ...

This could be implemented by adding some kind of `ParticleGroup` enum, and passing a `ParticleGroup` to `Compute` trait.

```rust
enum ParticlesGroup {
    All,
    Particles(Vec<usize>),
    Molecules(Vec<usize>),
}

// A simple selection language could be useful here
impl ParticlesGroup {
    pub fn all() -> ParticlesGroup;
    pub fn particles(selection: &str) -> ParticlesGroup;
    pub fn molecules(selection: &str) -> ParticlesGroup;
}
```

**Comment by g-bauer** [comment]
Do you want this as a part of `Output`?

**Comment by Luthaf** [comment]
No, my code example had no sense. This should not be part of the output, rather of the Compute trait.

This could be decomposed in a two-steps process:

1. Get the list of atoms in a given sub-group;
2. Use this list in all the [Compute](https://github.com/lumol-org/lumol/blob/75723f66e6b57ddb98856353d1536d5d4d2fc517/src/core/src/sys/compute.rs#L13-L18) trait implementation instead of the list of all atoms.
